### PR TITLE
Added several methods

### PR DESCRIPTION
**Pull request by SamPuttman**
Added most_accurate_team, least_accurate_team, most_tackles, and least_tackles. It was fairly simple after making the most_accurate_team, as I was able to copy most of that code into the other methods, only making simple changes.

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -239,6 +239,45 @@ def most_accurate_team
...
     # teams.csv - team_id, teamName
 
+  def least_accurate_team
+    season_data = {}
+        
+    @game_teams.each do |game_team|
+      game = find_game_by_id(game_team.game_id)
+      next unless game
+        
+      season = game.season
```
Comment (Chavezgm) [inline lib/stat_tracker.rb:553]: Nice job here! I really like how you assigned variables here !

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -239,6 +239,45 @@ def most_accurate_team
...
+      team_id = game_team.team_id
+      goals = game_team.goals.to_f
+      shots = game_team.shots.to_f
+            
+      # Initialize nested hash for each new season or team
+      season_data[season] ||= {}
```
Comment (Chavezgm) [inline lib/stat_tracker.rb:559]: I like how you used the ||= symbol

**Comment by dylanmade** [comment]
NOTE: Conflicts merged live by all three contributors

File: lib/stat_tracker.rb (hunk 1 above)
Comment (dylanmade) [reply lib/stat_tracker.rb:553]: Yeah this is amazing dude. You are a legend for pulling this off

File: lib/stat_tracker.rb (hunk 2 above)
Comment (dylanmade) [reply lib/stat_tracker.rb:559]: Yes. I had never seen this before

**Review by dylanmade — APPROVED** [review]

**Pull request merged**